### ISSUE TITLE
If using CMEK on Postgres and with Replicas, the instance is recreated every terraform run

### ISSUE THREAD

**Issue by jmymy** [terraform-google-modules/terraform-google-sql-db#283]
### TL;DR

If you set a CMEK on a sql database, and use a replica, the replica instance will be recreated every terraform apply.

### Expected behavior

The replica instance remains unchanged and does not need to be recreated

### Observed behavior

since the terraform module states to set the encryption_key_name to `null` for the replica, once its created the google api returns the value of the primary CMEK. Terraform code expects it to be `null`, so it recreates the replica instance (well trys to) with a null CMEK.

- trys to recreated since the replica name is basically hardcoded and you cant reuse a old name for 7 days. so you have to change the name each time, or wait 7 days to reuse the replica name again

### Terraform Configuration

```hcl
module "postgres-sql-db" {
  source              = "GoogleCloudPlatform/sql-db/google//modules/postgresql"
  version             = "10.0.0"
  name                = var.instance_name
  availability_type   = "REGIONAL"
  project_id          = var.project_id
  region              = var.region
  zone                = var.zones[0]
  database_version    = "POSTGRES_13"
  db_name             = var.db_name
  tier                = var.tier
  database_flags      = var.database_flags
  encryption_key_name = google_kms_crypto_key.pgsql.id
  module_depends_on   = var.module_depends_on
  ip_configuration = {
    ipv4_enabled        = true
    require_ssl         = var.require_ssl
    private_network     = var.private_network
    authorized_networks = var.authorized_networks
    allocated_ip_range  = null
  }
  maintenance_window_day          = 7
  maintenance_window_hour         = 12
  maintenance_window_update_track = "stable"
  backup_configuration = {
    enabled                        = true
    start_time                     = "20:55"
    point_in_time_recovery_enabled = true
    location                       = "us"
    retained_backups               = null
    retention_unit                 = null
    transaction_log_retention_days = null
  }
  read_replicas = [
    {
      name = var.pg_replica_name
      tier = var.tier
      zone = var.zones[0]
      ip_configuration = {
        ipv4_enabled        = true
        require_ssl         = var.require_ssl
        private_network     = var.private_network
        authorized_networks = var.authorized_networks
        allocated_ip_range  = null
      }
      database_flags      = var.database_replica_flags
      disk_autoresize     = true
      disk_size           = 10
      disk_type           = "PD_HDD"
      user_labels         = { bar = "baz" }
      encryption_key_name = null
      # encryption_key_name = google_kms_crypto_key.pgsql.id
    }
  ]
}
```

### Terraform Version

```sh
Terraform v1.1.7
```

### Additional information

change introduced here: https://github.com/terraform-google-modules/terraform-google-sql-db/pull/251

**Comment by imrannayer** [comment]
`encryption_key_name` is required to be null for replica in same region. If you provide `encryption_key_name` anything other than null value for replica in same region you will receive following error.  

`Error: Error, failed to create instance instance-name-1: googleapi: Error 400: Invalid request: Disk encryption config should not be specified for replica that is in the same region of primary.., invalid`

For a permanent fix either provider needs to ignore this value as a difference or API needs to handle this scenario.
For quick fix I can change module to ignore changes in  `encryption_key_name` using terraform lifecycle ignore_changes.

**Comment by imrannayer** [comment]
I have created an issue for provider to fix this behavior. 
[CloudSQL replica instance recreated when encryption key is passed as null #11270](https://github.com/hashicorp/terraform-provider-google/issues/11270)

**Comment by imrannayer** [comment]
Just noticed you already created the bug request. Closing mine.
